qutip/core/metrics.py

```diff
@@ -108,6 +108,35 @@ def fidelity(A, B):
     return float(np.real(np.sqrt(eig_vals[eig_vals > 0]).sum()))
 
 
+def _hilbert_space_dims(oper):
+    """
+    For a quantum channel `oper`, return the dimensions `[dims_out, dims_in]`
+    of the output Hilbert space and the input Hilbert space.
+    - If oper is a unitary, then `oper.dims == [dims_out, dims_in]`.
+    - If oper is a list of Kraus operators, then
+     `oper[0].dims == [dims_out, dims_in]`.
+    - If oper is a superoperator with `oper.superrep == 'super'`:
+     `oper.dims == [[dims_out, dims_out], [dims_in, dims_in]]`
+    - If oper is a superoperator with `oper.superrep == 'choi'`:
+     `oper.dims == [[dims_in, dims_out], [dims_in, dims_out]]`
+    - If oper is a superoperator with `oper.superrep == 'chi', then
+      `dims_out == dims_in` and
+      `oper.dims == [[dims_out, dims_out], [dims_out, dims_out]]`.
+    :param oper: A quantum channel, represented by a unitary, a list of Kraus
+    operators, or a superoperator
+    :return: `[dims_out, dims_in]`, where `dims_out` and `dims_in` are lists
+     of integers
+    """
+    if isinstance(oper, list):
+        return oper[0].dims
+    elif oper.type == 'oper':  # interpret as unitary quantum channel
+        return oper.dims
+    elif oper.type=='super' and oper.superrep in ['choi', 'chi', 'super']:
```

Comment:
```suggestion
    elif oper.type == 'super' and oper.superrep in ['choi', 'chi', 'super']:
```
Tiny style suggestion. We really should add black or something to our CI at some point (but not today).